### ISSUE TITLE
[Feature Contribution]: electricityMap Plugin 

### ISSUE THREAD

**Issue by yuu1127** [Green-Software-Foundation/carbon-aware-sdk#71]
### What happened?

This feature contribution will add electricityMap plugin support and various improvements to the Carbon Aware SDK WebAPI and CLI.  This plugin should be similar to Watttime plugin.
Electricity map supports more countries' data including many Asia such as India, China and others which Watttime doesn't support.  (#67 for details)
As discussed in a weekly meeting (5th July), target to merge into v1.1.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Feature Commitment

- [X] I commit to contributing this feature as a PR and working with the GSF to merge this feature into the Carbon Aware SDK.

**Comment by vaughanknight** [comment]
I'll create a branch for this as it may require collaboration across the project team and it will centralise it for review/collab across companies.

**Comment by vaughanknight** [comment]
Branch created here https://github.com/Green-Software-Foundation/carbon-aware-sdk/tree/feat/electricity-map

**Comment by vaughanknight** [comment]
Fork the branch into your own github repo, and work there.  But then PR into the electricity-map branch when you think it's complete.

**Comment by luwangshell** [comment]
How will the electricityMap license be handled?

**Comment by yuu1127** [comment]
@luwangshell,
I assume everyone uses the personal version and their token so far.
(I'm not sure this is the exact answer to your question ...)

**Comment by Kongkille** [comment]
Hi there, I'm an engineer working at Electricity Maps and also a maintainer of our public open-source repository (https://github.com/electricitymap/electricitymap-contrib)

Just wanted to pop in and say that I'm available in case you have any questions!

**Comment by YaSuenag** [comment]
Fundamental feature of electricityMap plugin has been merged https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/75

As @vaughanknight said in PR comment, it is not completed yet, we need to work further. It may happen after #95 .

Anyway we'd like to involve to electricityMap plugin because it covers wider area includes Asia. I think we should not close this issue until the work is done. I believe we can complete to work for this plugin in v1.1 !

**Comment by vaughanknight** [comment]
As we want to ensure everyone on the planet as much as possible can leverage the Carbon Aware SDK and public API, we should look to having ElectricityMap support as a priority in the next release, otherwise a large portion of the countries in the world will not be supported by the API.  As such this is a priority for v1.1.

**Comment by YaSuenag** [comment]
Agree with @vaughanknight !

Also NTT DATA wants this feature because Japan is not covered in WattTIme - I'm sure this feature helps all of green soft engineers in any other countries / regions (e.g. APAC, South America and more!) .

**Comment by YaSuenag** [comment]
Note that #194 supports electricityMap payed account only. We need to consider for free account, then we continue to discuss about it.
